src/georaster-layer-for-leaflet.ts

```diff
@@ -308,11 +328,12 @@ const GeoRasterLayer = L.GridLayer.extend({
                 values = tileRasters.map(band => band[h][w]);
               } else if (rasters) {
                 // get value from array with data for entire raster
-                values = rasters.map(band => {
+                values = rasters.map((band: Georaster) => {
+                  // @ts-ignore it works but needs validation. Current error: Type 'null' cannot be used as an index type
```

Comment:
That didn't work unfortunately. This block above it:

```javascript
         let yInRasterPixels: number | null = null;
          if (inSimpleCRS || this.projection === EPSG4326) {
            yInRasterPixels = Math.floor((yMaxOfLayer - lat) / pixelHeight);
          } else {
            yInRasterPixels = null;
          }
```

Is explicitly setting null in the `else`. 

We then drop out the `if` block and there is no check in the `for (let w = 0; w < numberOfSamplesAcross; w++) {` loop testing if `yInRasterPixels` is null. From the compilers point of view it's still possible that it's null and being used to index and array. That's what fails. 

It could be solve by moving that `for` loop into the `if` statement. Or defaulting `yInRasterPixels` to 0. Any ideas?

Reply:
Let's just set yInRasterPixels to 0.  I'm definitely open to optimizing/refactoring the logic of where to put the for loops later, but don't want to postpone merging your PR.